File: 문제풀이/온라인 저지/LeetCode/# 01. Easy/# 0001. Two Sum/python/solution3.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_map = {num: idx for idx, num in enumerate(nums)}

        for i, num in enumerate(nums):
            if target - num in nums_map and i != nums_map[target - num]:
                return [i, nums_map[target-num]]

File: 문제풀이/온라인 저지/LeetCode/# 01. Easy/# 0001. Two Sum/kotlin/Solution3.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val items = nums.mapIndexed{idx, value -> Data(idx, value)}.sorted()
        var lt = 0
        var rt = items.lastIndex
        var sum: Int
        while (lt < rt) {
            sum = items[lt].value + items[rt].value
            if (sum == target) {
                return if (items[lt].idx < items[rt].idx) intArrayOf(items[lt].idx , items[rt].idx)
                else intArrayOf(items[rt].idx, items[lt].idx)
            } else if (sum > target) {
                rt --
            } else {
                lt ++
            }
        }
        return intArrayOf()
    }
}

class Data(val idx: Int, val value: Int) : Comparable<Data> {
    override fun compareTo(other: Data): Int {
        return value - other.value
    }
}
